compiler/testData/diagnostics/tests/effects/controlflow/flowInlining/unreachableCode.kt

```diff
@@ -0,0 +1,11 @@
+// !DIAGNOSTICS: -INVISIBLE_MEMBER -INVISIBLE_REFERENCE -UNUSED_PARAMETER
+// !LANGUAGE: +CalledInPlaceEffect
+
+import kotlin.internal.*
+
+inline fun myRun(@CalledInPlace(InvocationCount.EXACTLY_ONCE) block: () -> Unit) = block()
+
+fun throwInLambda(): Int {
+    val x = myRun { throw java.lang.IllegalArgumentException() }
+    return <!TYPE_MISMATCH!>x<!>
```

Comment:
Sorry, that's erroneous overwrite of previous commit, i.e. this test data is wrong and test fails. `UNREACHABLE_CODE` is indeed reported here, and this diff shouldn't be in commit with introduction of language features.
As for type mismatch, we currently have the same behavior for ordinary lambdas.